tests/helper/helper_generic.go

```diff
@@ -162,6 +183,35 @@ func WatchNonRetCmdStdOut(cmdStr string, timeout time.Duration, check func(outpu
 	}
 }
 
+var randMutex = &sync.Mutex{}
```

Comment:
Can we not use global mutex? Consider having a struct with the mutex and the object it protects?